WA3CRUD/WA3CRUD/Models/Market.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace WA3CRUD.Models
+{
+    public class Market : IMarketable
+    {
+        private List<Product> products = new List<Product>();
+        private int _nextId = 1;
```

Comment:
use GUID for ids and you'll never need to care about logic for creating new unique ID